lib/galaxy/schema/invocation.py

```diff
@@ -218,3 +253,289 @@ class InvocationMessageResponseModel(BaseModel):
+        title="Workflow ID", description="The encoded Workflow ID associated with the invocation."
+    )
+    history_id: EncodedDatabaseIdField = Field(
+        default=Required,
+        title="History ID",
+        description="The encoded ID of the history associated with the invocation.",
+    )
+    # The uuid version here is 1, which deviates from the other UUIDs used as they are version 4.
+    uuid: Optional[Union[UUID4, UUID1]] = Field(
+        default=None, title="UUID", description="Universal unique identifier of the workflow invocation."
+    )
+    state: InvocationState = Field(
+        default=Required, title="Invocation state", description="State of workflow invocation."
+    )
+    model_class: schema.INVOCATION_MODEL_CLASS = ModelClassField(schema.INVOCATION_MODEL_CLASS)
+    steps: List[InvocationStep] = Field(
+        default=Required, title="Steps", description="Steps of the workflow invocation."
+    )
+    inputs: Dict[str, InvocationInput] = Field(
+        default=Required, title="Inputs", description="Input datasets/dataset collections of the workflow invocation."
+    )
+    input_step_parameters: Dict[str, InvocationInputParameter] = Field(
+        default=Required, title="Input step parameters", description="Input step parameters of the workflow invocation."
+    )
+    outputs: Dict[str, InvocationOutput] = Field(
+        default=Required, title="Outputs", description="Output datasets of the workflow invocation."
+    )
+    output_collections: Dict[str, InvocationOutputCollection] = Field(
+        default=Required,
+        title="Output collections",
+        description="Output dataset collections of the workflow invocation.",
+    )
+    output_values: Dict[str, Any] = Field(
+        default=Required, title="Output values", description="Output values of the workflow invocation."
+    )
+    messages: List[InvocationMessageResponseUnion] = Field(
+        default=Required,
+        title="Messages",
+        description="A list of messages about why the invocation did not succeed.",
+    )
+
+
+class InvocationJobsSummaryBaseModel(Model):
+    id: EncodedDatabaseIdField = InvocationIdField
+    states: Dict[JobState, int] = Field(
+        default=Required, title="States", description="The states of all the jobs related to the Invocation."
+    )
+    populated_state: JobState = Field(
+        default=Required,
+        title="Populated state",
+        description="The absolute state of all the jobs related to the Invocation.",
+    )
+
+
+class InvocationJobsResponse(InvocationJobsSummaryBaseModel):
+    model: schema.INVOCATION_MODEL_CLASS = ModelClassField(schema.INVOCATION_MODEL_CLASS)
+
+
+class InvocationStepJobsResponse(InvocationJobsSummaryBaseModel):
+    # TODO - this can also be Job or ImplicitCollectionJobs - how to handle this?
+    # model: schema.INVOCATION_STEP_MODEL_CLASS = ModelClassField(schema.INVOCATION_STEP_MODEL_CLASS)
+    model: Any
```

Comment:
You usually resolve this using `Union` instead of just one model, In is that what you need?